Scrapping_part_3/scrapping_part3/scrapping_part3/middlewares.py

```diff
@@ -0,0 +1,46 @@
+from stem import Signal
+from stem.control import Controller
+from scrapy.downloadermiddlewares.httpproxy import HttpProxyMiddleware
+from stem.util.log import get_logger
+
+from .settings import REQUESTS_PER_SAME_TOR_IDENTITY
+
+
+# Number of requests sent during the current tor identity
+NUM_SENT_REQUESTS = 0
```

Comment:
Bad practice to use a global variable when its not needed. This should be used as ProxyMiddleware instance attribute.

Reply:
all fixed